### PR TITLE
[GP-6] Trash handling for shows

### PR DESCRIPTION
**Pull request by sc0ttkclark**
I did this for fun. Consider it a community contribution. :)

[GP-6]

Origin:
https://wordpress.org/support/topic/gig-press-viewing-trash/

[GP-6]: https://moderntribe.atlassian.net/browse/GP-6

### CODE REVIEW

File: admin/handlers.php [hunk 1]
```diff
@@ -1006,6 +1006,28 @@ function gigpress_empty_trash() {
...
+	$wpdb->show_errors();
+	check_admin_referer('gigpress-action');
+
+	$show = [ 'show_status' => 'active' ];
+	$where = array( 'show_id' => absint( $_GET['show_id'] ) );
+
+	$restore_show = $wpdb->update( GIGPRESS_SHOWS, $show, $where );
+
+	if( $restore_show ) { ?>
+		<div id="message" class="updated fade"><p><?php esc_html_e( 'The selected show has been restored.', 'gigpress' ); ?></p></div>
```
Comment (juanfra) [inline admin/handlers.php:1024]: Maybe move the html outside the conditional or set the string into a var via ternary and display it via variable without the if?

File: admin/handlers.php [hunk 2]
```diff
@@ -1006,6 +1006,28 @@ function gigpress_empty_trash() {
...
+// HANDLER: RESTORE SHOW
+// ======================
+
+function gigpress_restore_show() {
+
+	global $wpdb;
+	$wpdb->show_errors();
+	check_admin_referer('gigpress-action');
+
+	$show = [ 'show_status' => 'active' ];
```
Comment (juanfra) [inline admin/handlers.php:1018]: Did we update the PHP version on this plugin? Should we use the short syntax?

File: admin/shows.php [hunk 3]
```diff
@@ -21,13 +21,19 @@ function gigpress_admin_shows() {
 		require_once('handlers.php');
 		gigpress_empty_trash();		
 	}	
-	
+	if(isset($_GET['gpaction']) && $_GET['gpaction'] == "restore") {
```
Comment (juanfra) [inline admin/shows.php:24]:
```suggestion
	if ( isset( $_GET['gpaction'] ) && 'restore' == $_GET['gpaction'] ) {
```

File: admin/shows.php [hunk 4]
```diff
@@ -21,13 +21,19 @@ function gigpress_admin_shows() {
...
+		require_once('handlers.php');
```
Comment (juanfra) [inline admin/shows.php:25]:
```suggestion
		require_once( 'handlers.php' );
```

File: admin/shows.php [hunk 5]
```diff
@@ -134,15 +143,23 @@ function gigpress_admin_shows() {
...
 			echo('>' . __("All", "gigpress") . '</a> <span class="count">(' . $all	. ')</span> | </li>');
+
 			echo('<li><a href="' . admin_url('admin.php?page=gigpress-shows&amp;scope=upcoming') . '"');
 			if($scope == 'upcoming') echo(' class="current"');
 			echo('>' . __("Upcoming", "gigpress") . '</a> <span class="count">(' . $upcoming	. ')</span> | </li>');
+
 			echo('<li><a href="' . admin_url('admin.php?page=gigpress-shows&amp;scope=past') . '"');
 			if($scope == 'past') echo(' class="current"');
-			echo('>' . __("Past", "gigpress") . '</a> <span class="count">(' . $past	. ')</span></li>');
+			echo('>' . __("Past", "gigpress") . '</a> <span class="count">(' . $past	. ')</span> | </li>');
```
Comment (juanfra) [inline admin/shows.php:158]:
```suggestion
			echo('>' . esc_html__( 'Past', 'gigpress' ) . '</a> <span class="count">(' . $past . ')</span> | </li>');
```

File: admin/shows.php [hunk 6]
```diff
@@ -134,15 +143,23 @@ function gigpress_admin_shows() {
...
+
+    		echo('<li><a href="' . admin_url('admin.php?page=gigpress-shows&amp;scope=deleted') . '"');
+	    	if($scope == 'deleted') echo(' class="current"');
+		    echo('>' . __("Deleted", "gigpress") . '</a> <span class="count">(' . $deleted	. ')</span></li>');
```
Comment (juanfra) [inline admin/shows.php:162]:
```suggestion
	    	if ( $scope == 'deleted' ) echo( ' class="current"' );
		    echo('>' . esc_html__( 'Deleted', 'gigpress' ) . '</a> <span class="count">(' . $deleted . ')</span></li>');
```

File: admin/shows.php [hunk 7]
```diff
@@ -266,7 +283,11 @@ function gigpress_admin_shows() {
 					<td><?php echo $showdata['country']; ?></td>
 					<td><?php echo $showdata['tour']; ?></td>
 					<td class="gp-centre">
-						<a href="<?php echo admin_url('admin.php?page=gigpress&amp;gpaction=edit&amp;show_id='.$show->show_id); ?>" class="edit" title="<?php _e("Edit", "gigpress"); ?>"><?php _e("Edit", "gigpress"); ?></a>&nbsp;|&nbsp;<a href="<?php echo admin_url('admin.php?page=gigpress&amp;gpaction=copy&amp;show_id='. $show->show_id); ?>" class="edit" title="<?php _e("Copy", "gigpress"); ?>"><?php _e("Copy", "gigpress"); ?></a>
+                        <? if ( $scope == 'deleted' ) : ?>
```
Comment (juanfra) [inline admin/shows.php:286]:
```suggestion
                        <? if ( 'deleted' == $scope ) : ?>
```

File: admin/shows.php [hunk 8]
```diff
@@ -266,7 +283,11 @@ function gigpress_admin_shows() {
...
+                            <a href="<?php echo wp_nonce_url( admin_url( 'admin.php?page=gigpress-shows&amp;scope=deleted&amp;gpaction=restore&amp;show_id=' . $show->show_id ), 'gigpress-action' ); ?>" class="edit" title="<?php _e("Restore", "gigpress"); ?>"><?php _e("Restore", "gigpress"); ?></a>
+                        <?php else : ?>
+	    					<a href="<?php echo admin_url('admin.php?page=gigpress&amp;gpaction=edit&amp;show_id='.$show->show_id); ?>" class="edit" title="<?php _e("Edit", "gigpress"); ?>"><?php _e("Edit", "gigpress"); ?></a>&nbsp;|&nbsp;<a href="<?php echo admin_url('admin.php?page=gigpress&amp;gpaction=copy&amp;show_id='. $show->show_id); ?>" class="edit" title="<?php _e("Copy", "gigpress"); ?>"><?php _e("Copy", "gigpress"); ?></a>
```
Comment (juanfra) [inline admin/shows.php:289]:
```suggestion
	    					<a href="<?php echo admin_url( 'admin.php?page=gigpress&amp;gpaction=edit&amp;show_id='.$show->show_id ); ?>" class="edit" title="<?php esc_attr_e( 'Edit', 'gigpress' ); ?>"><?php esc_html_e( 'Edit', 'gigpress' ); ?></a>&nbsp;|&nbsp;<a href="<?php echo admin_url('admin.php?page=gigpress&amp;gpaction=copy&amp;show_id='. $show->show_id); ?>" class="edit" title="<?php esc_attr_e( 'Copy', 'gigpress' ); ?>"><?php esc_html_e( 'Copy', 'gigpress' ); ?></a>
```

File: admin/shows.php [hunk 9]
```diff
@@ -292,7 +313,9 @@ function gigpress_admin_shows() {
 		</table>
 		<div class="tablenav">
 			<div class="alignleft">
-				<input type="submit" value="<?php _e('Trash selected shows', 'gigpress'); ?>" class="button-secondary" /> &nbsp; 
+                <?php if ( $scope != 'deleted' ) : ?>
```
Comment (juanfra) [inline admin/shows.php:316]:
```suggestion
                <?php if ( 'deleted' != $scope ) : ?>
```

File: admin/shows.php [hunk 10]
```diff
@@ -292,7 +313,9 @@ function gigpress_admin_shows() {
...
+				<input type="submit" value="<?php _e('Trash selected shows', 'gigpress'); ?>" class="button-secondary" /> &nbsp;
```
Comment (juanfra) [inline admin/shows.php:317]:
```suggestion
				<input type="submit" value="<?php esc_attr_e( 'Trash selected shows', 'gigpress' ); ?>" class="button-secondary" /> &nbsp;
```

**Review by juanfra — DISMISSED** [review]
A few minor changes, looking great! thanks for the contribution! :lol:

File: admin/handlers.php (hunk 2 above)
Comment (sc0ttkclark) [reply admin/handlers.php:1018]:
```suggestion
	$show = array( 'show_status' => 'active' );
```

File: admin/handlers.php (hunk 1 above)
Comment (sc0ttkclark) [reply admin/handlers.php:1024]: Keeping as is for now to keep it simple

**Pull request merged**